src/analysis/utils/disk-utils.ts

```diff
@@ -16,17 +16,19 @@ import { Runtime } from 'src/libs/ajax/leonardo/models/runtime-models';
 import * as Utils from 'src/libs/utils';
 
 export const pdTypeFromDiskType = (type: GoogleDiskType): GooglePdType =>
-  Utils.switchCase(
+  Utils.switchCase<GoogleDiskType, GooglePdType | undefined>(
     type,
     [googlePdTypes.standard.value, () => googlePdTypes.standard],
     [googlePdTypes.balanced.value, () => googlePdTypes.balanced],
     [googlePdTypes.ssd.value, () => googlePdTypes.ssd],
     [
       Utils.DEFAULT,
-      () =>
-        console.error(`Invalid disk type: Should not be calling googlePdTypes.fromString for ${JSON.stringify(type)}`),
+      () => {
+        console.error(`Invalid disk type: Should not be calling googlePdTypes.fromString for ${JSON.stringify(type)}`);
+        return undefined;
+      },
     ]
-  );
+  ) as GooglePdType; // TODO: Remove cast
```

Comment:
This "log error and return undefined" for unexpected cases looks to be a pattern. However, the possible undefined isn't handled (the return type does not include undefined). I suspect a better way to handle these would be to throw an error for unexpected inputs. That would allow us to actually be sure that the return type is `GooglePdType` instead of `GooglePdType | undefined`.

Reply:
can you add this comment to the TODO here? I agree throwing an error and having a better type on this seems desirable

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4154/commits/135af723e9c68271245b88e7bbb08ca68bd42104.